kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/DeleteCommand.java

```diff
@@ -46,11 +51,24 @@ public int run() throws IOException {
         View view = Datasets.load(uriOrName);
         Preconditions.checkArgument(viewMatches(view.getUri(), uriOrName),
             "Resolved view does not match requested view: " + view.getUri());
-        view.deleteAll();
+
+        if(console.isDebugEnabled()){
+          for(PartitionView<?> pv: (Iterable<PartitionView<?>>) view.getCoveringPartitions()){
+            console.debug("Deleting partition {}", pv.getLocation());
```

Comment:
The partitions returned by `getCoveringPartitions` may contain values that aren't in the view. For example, if I have a view for all of today but I'm partitioning by months, I get a partition for all of October.

When deleting data, we check that the view we are deleting aligns with partitions: meaning that the view is exactly equal to the partitions we will delete, with no data in the partitions that doesn't match the view. Without doing that check here, we would report that more data would be deleted than actually is deleted.

Does that make more sense?